graylog2-web-interface/src/components/bootstrap/NavDropdown.jsx

```diff
@@ -49,23 +49,25 @@ const StyledNavDropdown = styled(BootstrapNavDropdown)`
   ${menuItemStyles}
 `;
 
-const NavDropdown = ({ inactiveTitle, title, ...props }) => {
+const NavDropdown = ({ title, inactiveTitle, badge: Badge, ...props }) => {
   const isActive = inactiveTitle ? inactiveTitle !== title : undefined;
 
   return (
     <StyledNavDropdown {...props}
-                       title={<NavItemStateIndicator>{title}</NavItemStateIndicator>}
+                       title={<NavItemStateIndicator>{Badge ? <Badge text={title} /> : title}</NavItemStateIndicator>}
                        active={isActive} />
   );
 };
 
 NavDropdown.propTypes = {
+  title: PropTypes.oneOfType([PropTypes.string, PropTypes.object]).isRequired,
   inactiveTitle: PropTypes.string,
-  title: PropTypes.node.isRequired,
+  badge: PropTypes.func,
 };
 
 NavDropdown.defaultProps = {
   inactiveTitle: undefined,
+  badge: undefined,
 };
 
 const ModifiedNavDropdown = styled(ModifiedBootstrapNavDropdown)`
```

Comment:
Since you are changing the component it would be nice to migrate it to TS in the same step.